### PR TITLE
Clever To-Do List app is ready for production

### PR DESCRIPTION
**Pull request by Shist**
1. Created basic structure of the project (with needed dependencies)
2. HTML file of the app was refactored (removed useless lines)
3. Added and tested global styles (reset, vars, mixins, etc.)
4. Created and registered custom <app-button> for whole app
5. Added some extra styles; prepared burger menu button
6. Added tasks list layouts with their styles
7. Added css class and props logic for active date items
8. Removed setup() functions, we will make everything in Options API
9. Created dynamic dates list (15 days before and after today)
10. Added feature with everlasting scrolling of dates
11. Added feature with selecting current date item
12. Dates logic (data, methods, computed) was moved to Vuex
13. Added label for current date based on state in Vuex
14. Firebase dependencies were installed
15. Added simple burger menu with link to sign-in; other pages prepared
16. Added feature with closing burger menu by clicking on dark space
17. Added layout of Sign-In page
18. Added layout of Sign-Up page
19. Added custom snackbar (toast) for different messages
20. Custom toast removed; added toast from vue3-toastify
21. Added signIn & signUp v-models via getters/setters for Vuex states/mutations
22. Added validation for email and password on SignUpPage
23. Added registration of new users to the firebase DB
24. Added authorization for users (by email and password)
25. Added clearing of currUser after Log Out
26. Added navigation guard which moves user to SignInPage if not authorized
27. State of currUser is now clearing in case of invalid email or password
28. Added dynamic loading of tasks for authorized users (without dates yet)
29. SignIn & SignUp were refactored with Firebase Authorization
30. Toasts logic was encapsulated in corresponding mixin
31. Added try/catch (with error toast) for cases of logout failure
32. Added mixin that can convert error codes into detailed messages
33. SignIn & SignUp forms are now clearing in case of success operation
34. Added more route guards: if user signed in, he will go to '/' anyway
35. Now, if existing, user uid is adding to Vuex from the start
36. Added route and component for not existing page (404)
37. Added label with current account (email) at top of burger menu
38. Now user SignUp creates doc (which id is userUid) with empty tasks arr
39. Optimization: commiting userUid and userEmail is now at 1 place only
40. Made loading of user tasks from DB (to console log for now)
41. Dates list on UI is now synchronized with firebase DB data
42. A little optimization done: now we don't throw all tasks arr to dates
43. Vuex module with dates was renamed everywhere as 'datesAndTasks'
44. Now tasks list of each user per each date is showing on UI
45. Now clicking on any task item leads to the dynamic root by id
46. Added loading of tasks (with toasts) in case of F5 on task page
47. Added loading of task data on its page (label, title, description)
48. Added basic styles for task page layout
49. Prepared page (with html skeleton) for creation of new tasks
50. Added styles for task creation page
51. Added 'left-arrow' UI component to the app
52. Added vuex module for task-creation-page data with its variables
53. Added feature with creating and uploading new task to Firebase DB
54. Added additional fetching of all tasks just after new task creation
55. Removed type='submit' from all form buttons as they use @click event
56. Added date input to edit page; all dates refactored to locale time
57. Prepared logic for editing of existing task
58. Prepared logic for deletion of tasks (with modal component)
59. Added feature of editing tasks from UI to Firebase DB
60. Added feature with deleting of tasks from UI to Firebase DB
61. Added labels for loading and error while fetching task by id
62. Added feature with editing the status of the tasks
63. Added validation for tasks creation and editing (title and description)
64. Removed <app-button></app-button> as it is useless
65. Added blocking of buttons and links while loading on SignIn & SignUp
66. Added blocking buttons while creating and uploading new task
67. Added link 'this project on GitHub' to the burger menu
68. routes were refactored with 'meta: {requiresAuth: ...}'
69. Added label with tip that we can shift-scroll horizontally (if > 768px)
70. Names of firestore fields were moved to .env file; prepared env template
71. Added some rules for firebase config files inside .gitignore
72. Days of dates were refactored to be good at all possible locales
73. Deleted annoying success toats of loading tasks; checked --> '/' page
74. Added special button to cancel editing of task if needed
75. Added description and documentation for the project

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+const firebaseApp = initializeApp({
+  apiKey: process.env.VUE_APP_API_KEY,
+  authDomain: process.env.VUE_APP_AUTH_DOMAIN,
+  projectId: process.env.VUE_APP_PROJECT_ID,
+  storageBucket: process.env.VUE_APP_STORAGE_BUCKET,
+  messagingSenderId: process.env.VUE_APP_MESSAGING_SENDER_ID,
+  appId: process.env.VUE_APP_APP_ID,
+  measurementId: process.env.VUE_APP_MEASUREMENT_ID,
+});
```
Comment (ghost) [inline src/main.js]: Перемести в отдельную функцию

File: src/components/UI/AppLeftArrow.vue [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<template>
+  <button class="left-arrow">
+    <slot></slot>
```
Comment (ghost) [inline src/components/UI/AppLeftArrow.vue]: убрать slot

File: src/pages/SignInPage.vue [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      set(newValue) {
+        this.$store.commit("signIn/setEmail", newValue);
+      },
+    },
+    password: {
+      get() {
+        return this.$store.state.signIn.password;
+      },
+      set(newValue) {
+        this.$store.commit("signIn/setPassword", newValue);
```
Comment (ghost) [inline src/pages/SignInPage.vue]: поменять на map...

File: src/store/datesAndTasksModule.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import { format } from "date-fns";
```
Comment (ghost) [inline src/store/datesAndTasksModule.js:1]: данный стор не нужен

**Comment by Shist** [comment]
=== Новые правки ===
1) Перемещён конфиг файербейза в отдельную функцию
2) Убран ненужный слот для компонента `<left-arrow />`
3) Добавлены мапперы (стейтовые и мутационные) для компутед (которые из сетеров и гетеров) для вью моделей SignIn и SignUp. Правда, позже они были удалены, потому что согласну пункту #4 нам незачем хранить состояния их инпутов в сторе Vuex.
4) Из хранилища Vuex были убраны все модули, которые не используются глобально. Они были перенесены в локальные состояния соответствующих компонентов.
5) Перенести все вызовы firebase-овских функций из firebaseModule в специально созданный services/firebase/index.js. Сам firebaseModule был переименован в userDataModule, там хранятся только данные, которые присылает Firebase.
6) На странице одного таска теперь подтягивается только этот таск (а не все таски вместе, как это было раньше). При перемещении юзера на главную страницу, подгружаются уже все таски.
7) Файл .env переименован на .env.example, в нём хранятся только настройки конфига для Firebase (имена таблиц и полей самой базы предполагаются захардкоженными в коде).
9) Создан .prettierrc файл для проекта, во всех файлах с кодом были сделаны отступы между логическими блоками

**Review by ghost — APPROVED** [review]

**Pull request merged**